codesign/lib/src/utils.dart

```diff
@@ -2,7 +2,42 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io';
+
 import 'package:args/args.dart';
+import 'package:process/process.dart';
+
+/// helper function to generate unique next IDs.
+int id = 0;
+int get nextId {
```

Comment:
Yeah this function is used to produce file names for generated folders. Hash code without override would produce something like 649043408, which I think is less intuitive than the 1~20 numbers produced by sequential generation, which is based on the order of artifacts being processed.

Or we can create a class that has fileSystemEntity as a subfield, and override the hash code function to produce a number based on the name of fileSystemEntity. I might be wrong, but I think this approach is a bit harder to keep track of the file names as compared to sequential ordering.

Reply:
Do these ids have a use for humans? I would avoid adding complexity here if we only care about uniqueness

Reply:
Yeah the sequential numbers would help me track the entities in test. Using hash code in file name requires us to retrieve the actual entity instead of a reference of it. 

For example, if I use `$entity.hashcode` as part of the filename in the generated folder. Later in tests, I would like to obtain `$entity.hashcode` to verify the match. To compute this hash code, I would need to retrieve the entity. But if I use the file path to retrieve this entity, (something like `filesystem.file(path)`), I would obtain a reference to the entity, but not the entity itself. And the (reference of entity)'s hash code is not the same as the (actual entity)'s hash code, so the file names would not match.

Reply:
Based on what you outlined, it sounds like a much more obvious choice to use hashcode. In the long term, hard coding these ids based on assumptions is going to create brittle tests. Granted, I would recommend instead of using the dart:io entity hashcode, but instead the hashcode of the path string.

Reply:
Thanks for explaining! done